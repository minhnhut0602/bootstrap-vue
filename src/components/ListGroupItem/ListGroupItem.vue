<template>
    <component
        :is="tag"
        class="list-group-item"
        :class="classes"
    >
        <slot />
    </component>
</template>

<script lang="ts" setup>
import useActive, {activeProps} from '../../composables/useActive';
import useDisabled, {disabledProps} from '../../composables/useDisabled';
import useVariant, {variantProps} from '../../composables/useVariant';
import {computed} from 'vue';
import useClasses from '../../composables/useClasses';

const props = defineProps({
    ...activeProps,
    ...disabledProps,
    tag: {
        type: String,
        default: 'li',
    },
    variant: {
        ...variantProps.variant,
        default: '',
    },
});


const {classes} = useClasses([
    computed(() => ['a', 'button'].includes(props.tag) ? 'list-group-item-action' : undefined).value,
    useActive(props.active).activeClass.value,
    useDisabled(props.disabled).disabledClass.value,
    useVariant(props.variant, 'list-group-item-{0}').variantClass.value
]);
</script>
