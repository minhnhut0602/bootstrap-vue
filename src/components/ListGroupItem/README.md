# ListGroupItem

```vue live
<ListGroupItem>Default Example Usage</ListGroupItem>
```

## Props

| Prop name | Description | Type            | Values | Default |
| --------- | ----------- | --------------- | ------ | ------- |
| tag       |             | string          | -      | 'li'    |
| variant   |             | variantProp('') | -      |         |

## Slots

| Name    | Description | Bindings |
| ------- | ----------- | -------- |
| default |             |          |
