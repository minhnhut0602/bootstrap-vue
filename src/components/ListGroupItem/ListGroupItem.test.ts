import ListGroupItem from '.';
import {shallowMount} from '@vue/test-utils';

describe('template', () => {
    componentRenderTest(ListGroupItem);

    componentWrapperClassTest(
        ListGroupItem,
        {tag: 'a'},
        'list-group-item-action',
        'adds list-group-item-class when tag is a',
    );

    componentWrapperClassTest(
        ListGroupItem,
        {tag: 'button'},
        'list-group-item-action',
        'adds list-group-item-class when tag is button',
    );

    componentWrapperClassTest(ListGroupItem, {active: true}, 'active');

    componentWrapperClassTest(ListGroupItem, {variant: 'danger'}, 'list-group-item-danger');

    it('uses given tag', () => {
        const wrapper = shallowMount(ListGroupItem, {
            props: {tag: 'table'},
        });

        expect((wrapper.element as HTMLElement).tagName).toBe('TABLE');
    });
});
