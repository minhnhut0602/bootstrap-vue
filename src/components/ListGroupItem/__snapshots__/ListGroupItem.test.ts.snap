// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `
"<li class=\\"list-group-item list-group-item-\\">
  <!-- @slot Displays the content of the list group item -->
</li>"
`;
